crates/tests/nightly_component/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+mod bindings;
+use std::mem::*;
+use std::sync::*;
+use windows::core::*;
+use windows::Win32::Foundation::*;
+
+#[implement(bindings::Class)]
+struct Class(RwLock<i32>);
```

Comment:
Using [`AtomicI32`](https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicI32.html) instead of `RwLock<i32>` is preferable.